cypress/support/cypress-support/src/utils.js

```diff
@@ -443,8 +443,17 @@ function assertWithRequirementLogs(pretext, actual, expected, equateDeep = false
       assert(false, pretext + ': ' + JSON.stringify(errorObject));
     });
   } else {
+    let expectedLog = expected;
+    let actualLog = actual;
+    if (Array.isArray(actual) && actual.length < 1) {
+      actualLog = JSON.stringify(actual);
+    }
+    if (Array.isArray(expected) && expected.length < 1) {
+      expectedLog = JSON.stringify(expected);
+    }
+
     cy.log(
```

Comment:
Use firelog

Reply:
replaced assert with firelog

Reply:
Firelog will do the logging, you don't have to cy.log explicitly. That's the entire reason behind the review comment.

It should be 

```
let logMessage = pretext + ': Expected : ' + expectedLog + ' , Actual : ' + actualLog
equateDeep ? fireLog.deepEqual(actual, expected, logMessage) : fireLog.equal(actual, expected, logMessage);
```

Use if else instead of ternary operator if desired.

Reply:
Updated log as :
  
    const logMessage = pretext + ': Expected : ' + expectedLog + ' , Actual : ' + actualLog;
    if (equateDeep) {
      fireLog.deepEqual(actual, expected, logMessage);
    } else {
      fireLog.equal(actual, expected, logMessage);
    }